examples/text-classification/run_glue.py

```diff
@@ -134,24 +134,37 @@ def main():
     )
 
     # Get datasets
-    train_dataset = GlueDataset(data_args, tokenizer=tokenizer) if training_args.do_train else None
-    eval_dataset = GlueDataset(data_args, tokenizer=tokenizer, mode="dev") if training_args.do_eval else None
-    test_dataset = GlueDataset(data_args, tokenizer=tokenizer, mode="test") if training_args.do_predict else None
+    train_dataset = (
+        GlueDataset(data_args, tokenizer=tokenizer, cache_dir=model_args.cache_dir) if training_args.do_train else None
+    )
+    eval_dataset = (
+        GlueDataset(data_args, tokenizer=tokenizer, mode="dev", cache_dir=model_args.cache_dir)
+        if training_args.do_eval
+        else None
+    )
+    test_dataset = (
+        GlueDataset(data_args, tokenizer=tokenizer, mode="test", cache_dir=model_args.cache_dir)
+        if training_args.do_predict
+        else None
+    )
+
+    def build_compute_metrics_fn(task_name: str) -> Dict:
+        def compute_metrics_fn(p: EvalPrediction):
+            if output_mode == "classification":
+                preds = np.argmax(p.predictions, axis=1)
+            elif output_mode == "regression":
+                preds = np.squeeze(p.predictions)
+            return glue_compute_metrics(task_name, preds, p.label_ids)
 
-    def compute_metrics(p: EvalPrediction) -> Dict:
-        if output_mode == "classification":
-            preds = np.argmax(p.predictions, axis=1)
-        elif output_mode == "regression":
-            preds = np.squeeze(p.predictions)
-        return glue_compute_metrics(data_args.task_name, preds, p.label_ids)
+        return compute_metrics_fn
 
     # Initialize our Trainer
     trainer = Trainer(
         model=model,
         args=training_args,
         train_dataset=train_dataset,
         eval_dataset=eval_dataset,
-        compute_metrics=compute_metrics,
+        compute_metrics=build_compute_metrics_fn(data_args.task_name),
```

Comment:
why did you need this? I'm fine with the other changes.

Reply:
`task_name = mnli` in line 157 which is then called by: https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L794 which is this function: https://github.com/huggingface/transformers/blob/master/src/transformers/data/metrics/__init__.py#L65-L68 but with `task_name` as `mnli` both whether its `mnli` or `mnli-mm` dataset. This was one way to address it.

Reply:
Yes, but `glue_compute_metrics` handles the case "mnli-mm" correctly right now, no?

Reply:
Whether it's `mnli` or `mnli-mm` they both take this code path: https://github.com/huggingface/transformers/blob/master/src/transformers/data/metrics/__init__.py#L67-L68